{
  "cve": {
    "data_type": "CVE",
    "data_format": "MITRE",
    "data_version": "4.0",
    "CVE_data_meta": {
      "ID": "CVE-2018-1000030",
      "ASSIGNER": "cve@mitre.org"
    },
    "affects": {
      "vendor": {
        "vendor_data": [
          {
            "vendor_name": "python",
            "product": {
              "product_data": [
                {
                  "product_name": "python",
                  "version": {
                    "version_data": [
                      {
                        "version_value": "2.7.14",
                        "version_affected": "<="
                      }
                    ]
                  }
                }
              ]
            }
          },
          {
            "vendor_name": "canonical",
            "product": {
              "product_data": [
                {
                  "product_name": "ubuntu_linux",
                  "version": {
                    "version_data": [
                      {
                        "version_value": "12.04",
                        "version_affected": "="
                      },
                      {
                        "version_value": "14.04",
                        "version_affected": "="
                      },
                      {
                        "version_value": "16.04",
                        "version_affected": "="
                      },
                      {
                        "version_value": "18.04",
                        "version_affected": "="
                      }
                    ]
                  }
                }
              ]
            }
          }
        ]
      }
    },
    "problemtype": {
      "problemtype_data": [
        {
          "description": [
            {
              "lang": "en",
              "value": "CWE-119"
            },
            {
              "lang": "en",
              "value": "CWE-416"
            }
          ]
        }
      ]
    },
    "references": {
      "reference_data": [
        {
          "url": "https://bugs.python.org/issue31530",
          "name": "https://bugs.python.org/issue31530",
          "refsource": "CONFIRM",
          "tags": [
            "Issue Tracking",
            "Patch",
            "Vendor Advisory"
          ]
        },
        {
          "url": "https://drive.google.com/file/d/1oyR9DAZjZK_SCn3mor6NRAYLJS6ueXaY/view",
          "name": "https://drive.google.com/file/d/1oyR9DAZjZK_SCn3mor6NRAYLJS6ueXaY/view",
          "refsource": "MISC",
          "tags": [
            "Third Party Advisory"
          ]
        },
        {
          "url": "https://security.gentoo.org/glsa/201811-02",
          "name": "GLSA-201811-02",
          "refsource": "GENTOO",
          "tags": [
            "Third Party Advisory"
          ]
        },
        {
          "url": "https://usn.ubuntu.com/3817-1/",
          "name": "USN-3817-1",
          "refsource": "UBUNTU",
          "tags": [
            "Third Party Advisory"
          ]
        },
        {
          "url": "https://usn.ubuntu.com/3817-2/",
          "name": "USN-3817-2",
          "refsource": "UBUNTU",
          "tags": [
            "Third Party Advisory"
          ]
        },
        {
          "url": "https://www.dropbox.com/sh/sj3ee7xv55j36k7/AADwP-YfOYikBMuy32e0uvPFa?dl=0",
          "name": "https://www.dropbox.com/sh/sj3ee7xv55j36k7/AADwP-YfOYikBMuy32e0uvPFa?dl=0",
          "refsource": "MISC",
          "tags": [
            "Permissions Required"
          ]
        }
      ]
    },
    "description": {
      "description_data": [
        {
          "lang": "en",
          "value": "Python 2.7.14 is vulnerable to a Heap-Buffer-Overflow as well as a Heap-Use-After-Free. Python versions prior to 2.7.14 may also be vulnerable and it appears that Python 2.7.17 and prior may also be vulnerable however this has not been confirmed. The vulnerability lies when multiply threads are handling large amounts of data. In both cases there is essentially a race condition that occurs. For the Heap-Buffer-Overflow, Thread 2 is creating the size for a buffer, but Thread1 is already writing to the buffer without knowing how much to write. So when a large amount of data is being processed, it is very easy to cause memory corruption using a Heap-Buffer-Overflow. As for the Use-After-Free, Thread3->Malloc->Thread1->Free's->Thread2-Re-uses-Free'd Memory. The PSRT has stated that this is not a security vulnerability due to the fact that the attacker must be able to run code, however in some situations, such as function as a service, this vulnerability can potentially be used by an attacker to violate a trust boundary, as such the DWF feels this issue deserves a CVE."
        }
      ]
    }
  },
  "configurations": {
    "CVE_data_version": "4.0",
    "nodes": [
      {
        "operator": "OR",
        "cpe_match": [
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
            "versionEndIncluding": "2.7.14"
          }
        ]
      },
      {
        "operator": "OR",
        "cpe_match": [
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:o:canonical:ubuntu_linux:12.04:*:*:*:esm:*:*:*"
          },
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:o:canonical:ubuntu_linux:14.04:*:*:*:lts:*:*:*"
          },
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:o:canonical:ubuntu_linux:16.04:*:*:*:lts:*:*:*"
          },
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:o:canonical:ubuntu_linux:18.04:*:*:*:lts:*:*:*"
          }
        ]
      }
    ]
  },
  "impact": {
    "baseMetricV3": {
      "cvssV3": {
        "version": "3.0",
        "vectorString": "CVSS:3.0/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
        "attackVector": "NETWORK",
        "attackComplexity": "HIGH",
        "privilegesRequired": "NONE",
        "userInteraction": "NONE",
        "scope": "UNCHANGED",
        "confidentialityImpact": "HIGH",
        "integrityImpact": "HIGH",
        "availabilityImpact": "HIGH",
        "baseScore": 8.1,
        "baseSeverity": "HIGH"
      },
      "exploitabilityScore": 2.2,
      "impactScore": 5.9
    },
    "baseMetricV2": {
      "cvssV2": {
        "version": "2.0",
        "vectorString": "AV:N/AC:M/Au:N/C:P/I:P/A:P",
        "accessVector": "NETWORK",
        "accessComplexity": "MEDIUM",
        "authentication": "NONE",
        "confidentialityImpact": "PARTIAL",
        "integrityImpact": "PARTIAL",
        "availabilityImpact": "PARTIAL",
        "baseScore": 6.8
      },
      "severity": "MEDIUM",
      "exploitabilityScore": 8.6,
      "impactScore": 6.4,
      "acInsufInfo": false,
      "obtainAllPrivilege": false,
      "obtainUserPrivilege": false,
      "obtainOtherPrivilege": false,
      "userInteractionRequired": false
    }
  },
  "publishedDate": "2018-02-08T17:29Z",
  "lastModifiedDate": "2019-03-26T18:18Z"
}